src/applications/facility-locator/components/LatencyIndicator.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+
+class LatencyIndicator extends React.Component {
+  static defaultProps = {
+    latencyThreshold: 2000,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      sufferingLatency: false,
+    };
+  }
+
+  componentDidMount() {
+    this.timeoutId = setTimeout(() => {
+      this.setState({ sufferingLatency: true });
+    }, this.props.latencyThreshold);
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timeoutId);
+  }
+
+  render() {
+    return (
+      <AlertBox
```

Comment:
Could use a render prop here and make this component generic view-behavior related instead of app-concern related. Like “delayed render” or something.

Reply:
Is there a place in `src/platform` where it would make sense to add this?

Reply:
`src/platform/utilities/ui`, probably. That's where our async loader component lives.